src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -448,9 +448,9 @@ LanguageExpression getResourceInfoExpression()
 
                 case "outputs":
                     var moduleSymbol = reference.Module;
-
                     if (context.SemanticModel.Features.SecureOutputsEnabled &&
-                        FindPossibleSecretsVisitor.FindPossibleSecretsInExpression(context.SemanticModel, moduleSymbol.DeclaringModule).Any())
+                        expression.SourceSyntax is not null &&
+                        FindPossibleSecretsVisitor.FindPossibleSecretsInExpression(context.SemanticModel, expression.SourceSyntax).Any())
```

Comment:
Shouldn't we just need to check whether the output has a `@secure()` decorator on it, rather than using this linter rule?

Reply:
See https://github.com/Azure/bicep/blob/8cfcfa3affd928a6df75a4f0db2958c902d792df/src/Bicep.Core/Semantics/SymbolExtensions.cs#L39 for an example of checking whether a parameter symbol has a `@secure()` decorator.

Reply:
ARM will redact any output that *contains* a secure value from the normal `reference()` output. For example,

```bicep
@secure()
type secret = string
type secrets = secret[]

output secrets secrets = [<some sensitive value>]
```

will cause the `secrets` output to be accessible over the `listOutputsWithSecureValues()` function but not via the `reference()` function.

Reply:
Thanks for the explanation!